src/puppetlabs/trapperkeeper/services/status/status_proxy_service.clj

```diff
@@ -0,0 +1,22 @@
+(ns puppetlabs.trapperkeeper.services.status.status-proxy-service
+  (:require [clojure.tools.logging :as log]
+    [puppetlabs.trapperkeeper.core :refer [defservice]])
+  (:import (java.net URL)))
+
+(defservice status-proxy-service
+  [[:WebroutingService add-proxy-route]
+   [:ConfigService get-in-config]]
+  (init [this context]
+    (log/info "Initializing status service proxy")
+    (let [target-url (URL. (get-in-config [:status-proxy :proxy-target-url]))
+          host (.getHost target-url)
+          port (.getPort target-url)
+          path (.getPath target-url)]
+      (add-proxy-route
+        this
+        {:host host
+         :port port
+         :path path}
+        {:ssl-config (get-in-config [:status-proxy :ssl-opts])
```

Comment:
Hmmm, actually defaulting to `:use-server-config` probably isn't going to make sense for most uses of the service anyway since the status proxy would typically be registered with a Jetty server instance which probably wouldn't have any SSL configuration associated with it because it would only have a plaintext port configured for it.

Maybe it would be better to do some validation in here to confirm that the `ssl-opts` is non-empty and matches an expected schema rather than deferring to Jetty to validate this upstream and produce a potentially less useful failure message, though.